Fixing a few complaints by https://github.com/drwetter/testssl.sh
    1. Should be using SHA284 for secp384r1 (curve P-384) keys
    2. Serial number needs a minum length 8 bytes
    3. Certificate Validity is to not exceed 398 days
    4. Changed Phoenix Technologies Ltd. to Phoenix Technologies Inc.

diff --git a/include/ssl_key_handler.hpp b/include/ssl_key_handler.hpp
index 6a9bd98..0bda2c2 100644
--- a/include/ssl_key_handler.hpp
+++ b/include/ssl_key_handler.hpp
@@ -240,17 +240,18 @@ inline void generateSslCertificate(const std::string& filepath,
             // number If this is not random, regenerating certs throws broswer
             // errors
             bmcweb::OpenSSLGenerator gen;
-            std::uniform_int_distribution<int> dis(
-                1, std::numeric_limits<int>::max());
-            int serial = dis(gen);
+            std::uniform_int_distribution<unsigned long long> dis(
+                1, std::numeric_limits<unsigned long long>::max());
+            unsigned long long int serial = dis(gen);
 
-            ASN1_INTEGER_set(X509_get_serialNumber(x509), serial);
+            // length should be >= 64 bits entropy
+            ASN1_INTEGER_set_uint64(X509_get_serialNumber(x509), serial);
 
             // not before this moment
             X509_gmtime_adj(X509_get_notBefore(x509), 0);
-            // Cert is valid for 10 years
+            // Cert is valid for 398 days.
             X509_gmtime_adj(X509_get_notAfter(x509),
-                            60L * 60L * 24L * 365L * 10L);
+                            60L * 60L * 24L * 398L);
 
             // set the public key to the key we just generated
             X509_set_pubkey(x509, pPrivKey);
@@ -264,7 +265,7 @@ inline void generateSslCertificate(const std::string& filepath,
                 reinterpret_cast<const unsigned char*>("US"), -1, -1, 0);
             X509_NAME_add_entry_by_txt(
                 name, "O", MBSTRING_ASC,
-                reinterpret_cast<const unsigned char*>("Phoenix Technologies Ltd."), -1, -1, 0);
+                reinterpret_cast<const unsigned char*>("Phoenix Technologies Inc."), -1, -1, 0);
             X509_NAME_add_entry_by_txt(
                 name, "CN", MBSTRING_ASC,
                 reinterpret_cast<const unsigned char*>(cn.c_str()), -1, -1, 0);
@@ -281,7 +282,7 @@ inline void generateSslCertificate(const std::string& filepath,
             addExt(x509, NID_netscape_comment, (x509Comment));
 
             // Sign the certificate with our private key
-            X509_sign(x509, pPrivKey, EVP_sha256());
+            X509_sign(x509, pPrivKey, EVP_sha384());
 
             pFile = fopen(filepath.c_str(), "wt");
 
@@ -371,6 +372,7 @@ inline std::shared_ptr<boost::asio::ssl::context>
     // unauthenticated endpoints
     // mSslContext->set_verify_mode(boost::asio::ssl::verify_peer);
 
+    SSL_CTX_set_options(mSslContext->native_handle(), SSL_OP_NO_RENEGOTIATION);
     SSL_CTX_set_options(mSslContext->native_handle(), SSL_OP_CIPHER_SERVER_PREFERENCE);
 
     BMCWEB_LOG_DEBUG << "Using default TrustStore location: " << trustStorePath;
